sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java

```diff
@@ -94,6 +94,49 @@ public boolean isAccessible() {
   }
 
   /**
+   * {@link NestedValueProvider} is an implementation of {@link ValueProvider} that
+   * allows for wrapping another {@link ValueProvider} object.
+   */
+  public static class NestedValueProvider<T, X> implements ValueProvider<T>, Serializable {
+
+    /**
+     * An interface that provides a translator for a {@link NestedValueProvider}
+     * to produce a value from a deferred {@link ValueProvider}.
```

Comment:
we should pick some consistent logic about these names. Are they "deferrable" or "runtime" or something else?